src/distilabel/pipeline/templates/instruction.py

```diff
@@ -0,0 +1,136 @@
+    MagpieGenerator and responses for a given system prompt. The pipeline then keeps only
+    the instruction, response, and model_name columns.
+
+    References:
+        - [Magpie: Alignment Data Synthesis from Scratch by Prompting Aligned LLMs with Nothing](https://arxiv.org/abs/2406.08464)
+
+    Example:
+
+        Generate instructions and responses for a given system prompt:
+
+        ```python
+        from distilabel.pipeline import InstructionResponsePipeline
+
+        pipeline = InstructionResponsePipeline()
+
+        distiset = pipeline.run()
+        ```
+
+        Customizing the pipeline further:
+
+        ```python
+        from distilabel.pipeline import InstructionResponsePipeline
+
+        pipeline = InstructionResponsePipeline(
+            system_prompt="You are a creative AI Assistant for writing science fiction.",
+            llm=InferenceEndpointsLLM(
+                model_id="meta-llama/Meta-Llama-3.2-3B-Instruct",
+                tokenizer_id="meta-llama/Meta-Llama-3.2-3B-Instruct",
+                generation_kwargs={"max_new_tokens": 512, "temperature": 0.7},
+            ),
+            num_rows=500,
+            batch_size=2,
+            n_turns=2,
+        )
+        ```
+    """
+
+    def __init__(
+        self,
+        llm: Optional[LLM] = None,
+        system_prompt: str = "You are a creative AI Assistant writer.",
+        hf_token: Optional[str] = None,
+        n_turns: int = 1,
+        num_rows: int = 10,
+        batch_size: int = 1,
+    ) -> None:
+        if llm is None:
+            self.llm: LLM = InferenceEndpointsLLM(
+                model_id=MODEL,
+                tokenizer_id=MODEL,
+                magpie_pre_query_template="llama3",
+                generation_kwargs={
+                    "temperature": 0.9,
+                    "do_sample": True,
+                    "max_new_tokens": 2048,
+                    "stop_sequences": [
+                        "<|eot_id|>",
+                        "<|start_header_id|>",
+                        "assistant",
+                        " \n\n",
+                    ],
+                },
+                api_key=hf_token,
+            )
+        else:
+            if isinstance(
+                llm, InferenceEndpointsLLM
+            ) and llm.magpie_pre_query_template not in [
+                "llama3",
+                "qwen2",
+            ]:
+                raise ValueError(
+                    "The provided pre-query template is not supported by Magpie. Supported templates are 'llama3' and 'qwen2'."
+                )
+            self.llm = llm
+
+        self.pipeline: Pipeline = self._get_magpie_pipeline(
+            system_prompt=system_prompt,
+            n_turns=n_turns,
+            num_rows=num_rows,
+            batch_size=batch_size,
+        )
+
+    def run(self) -> Distiset:
+        """Runs the pipeline and returns a Distiset."""
+        return self.pipeline.run()
```

Comment:
```suggestion
    def run(self, **kwargs) -> Distiset:
        """Runs the pipeline and returns a Distiset."""
        return self.pipeline.run(**kwargs)
```
Maybe to avoid restricting the arguments allowed we can just let it generic here